base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs

```diff
@@ -1047,6 +1036,104 @@ impl OutputSql {
             .first::<OutputSql>(conn)?)
     }
 
+    /// Return the available, time locked, pending incoming and pending outgoing balance
+    pub fn get_balance(tip: Option<u64>, conn: &SqliteConnection) -> Result<Balance, OutputManagerStorageError> {
+        #[derive(QueryableByName, Clone)]
+        struct BalanceQueryResult {
+            #[sql_type = "diesel::sql_types::BigInt"]
+            amount: i64,
+            #[sql_type = "diesel::sql_types::Text"]
+            category: String,
+        }
+        let balance_query_result = if let Some(val) = tip {
+            let balance_query = sql_query(
+                "SELECT coalesce(sum(value), 0) as amount, 'available_balance' as category \
+                 FROM outputs WHERE status = ? \
```

Comment:
Should take the mined heights into account when the tip is specified
```suggestion
                 FROM outputs WHERE status = ?  and mined_height < $val and  deleted_at_height > $val\ 
```

Reply:
I think this is a job for the validation code though, I think it is cleaner that this function is based purely on the status.